Registry mvp docs (#689)
wandb/docs: docs/guides/registry/configure_registry.md

Reviewer: there a few important [callouts](https://weightsandbiases.slack.com/archives/C05UUJ350CA/p1712262412947029) that on roles and settings that would be good to review / decide if or where they should go in this guide:
<img width="448" alt="Screenshot 2024-06-05 at 7 46 40 PM" src="https://github.com/wandb/docodile/assets/40642416/73afd5ef-d2ff-4983-addc-b85948532d18">

Author: We need to find a way to make this easy to understand for new users....I suggest that some of this be doubly emphasized in the rollout plan/email/marketing. In like a "What's new" or "What's changing".  From that wording, we can steal that and add it on the landing page for the Registry docs.

Reviewer: we don't need to directly put this list - I just wanted to share it with you to give context on different concepts that are distributed already across this PR (its a mini FAQ for you to consult on some of the default behaviors)

---
displayed_sidebar: default
---

# Configure registry access
Registries, and the linked artifacts inside of them, are scoped at the organization level. This means that teams within an organization can share artifacts linked in a registry.

Organization admins can limit who can access a registry by configuring user-defined roles for that registry. User defined roles are useful for fine-grained control on how specific members within an organization can interact with a registry. 

:::tip
The term 'user' is an all encompassing term that refers to any person that uses W&B, independent of their role type.
:::

For example, a user with member access in the core Model registry can be be assigned viewer access in the core Dataset registry.
## Registry roles

A user within an organization can have different roles, and therefore permissions, for each registry in their organization. Organization administrators can set a given user's permission.

The proceeding table lists the different roles a user can have and their permissions:

> | Role | Permissions |

Author: @noaleetz Can you fill these in? :)

Reviewer: filled this out in this commit - https://github.com/wandb/docodile/commit/5f88d886ae481d5ba2fce6ee336ec6dfa6bca77f

---
displayed_sidebar: default
---

> # Configure registry access

Reviewer: we should just call this registry settings or something because we will refer to the registry settings across this doc

Author: Given how complex the access roles/rules are, it makes sense to have this page front and cnter. We can make a separate page for other settings?

Reviewer: Sure - what if we had 2 pages:
1. Access Controls
2. Registry Settings (separate page for other settings)

Author: This will be a follow up task: https://wandb.atlassian.net/browse/DOCS-888

> Registries, and the linked artifacts inside of them, are scoped at the organization level. This means that teams within an organization can share artifacts linked in a registry.

Reviewer: ```suggestion
Registries, and the linked artifacts inside of them, belong to the organization. This means that teams within an organization can publish and consume artifacts linked to a registry, given the correct access control.
```

> Organization admins can limit who can access a registry by configuring user-defined roles for that registry. User defined roles are useful for fine-grained control on how specific members within an organization can interact with a registry. 

Reviewer: nope - registry admins (by default an org admin will be a registry admin, but someone who is not an org admin can still be granted this access for a registry). 

the roles are not user defined - we have 4 options for roles - viewer, member, admin, owner

Author: A little confused, can't the admin decide these roles for members of an org?

Reviewer: Ah yes - so this part is a bit tricky - we have 3 set of roles (or 3 scopes of a role): Org roles, team roles, registry roles. Your role in a team has no impact or relationship on your role in any registry - a registry is an object that belongs to org, not a team. So no matter your role in any team, it won't impact any permissions you have on any team. 

Custom roles only exist for [teams](users can be assigned custom org roles only at the team level.) (you can't have a custom role inside an org (in [this](https://docs.wandb.ai/guides/hosting/iam/manage-users#invite-users) section of docs you'll see that when managing org users you only have these 3 options: "Select the role you want to assign to the user, from Admin, Member or Viewer. By default, all users are assigned a Member role". 

To be more specific - the language I was calling out here is that (1) Registry admins limit registry access rather than org admins (you can be a registry admin either by being an org admin, or if someone makes you an admin in the registry), and (2) you can't have a user-defined role in a registry - you can only be assigned 1 of 3 roles. 

> Organization admins can limit who can access a registry by configuring user-defined roles for that registry. User defined roles are useful for fine-grained control on how specific members within an organization can interact with a registry. 
Registry admins can limit who can access a registry by navigating to a registry's settings and assigning a user's role to Admin, Member, or Viewer. User's can have different roles in different registries, for example, a viewer in the Model Registry and a member in the Dataset Registry. This is useful to limit who has write-access to certain registries.

Author: Noted. I changed some text and created an info blurb to summarize this.

<img width="646" alt="Screen Shot 2024-06-25 at 10 13 04 PM" src="https://github.com/wandb/docodile/assets/15202580/074768bc-7ae5-46f3-98c1-b8bf38f3f91a">

As a follow up, we should create a diagram to show this. I suspect customers will be confused by this as well.

Author: https://wandb.atlassian.net/browse/DOCS-889

:::tip
The term 'user' is an all encompassing term that refers to any person that uses W&B, independent of their role type.
:::

> For example, a user with member access in the core Model registry can be be assigned viewer access in the core Dataset registry.

Reviewer: let's avoid referring to something as a core Dataset Registry - we should just call it Model Registry or Dataset Registry. There really isn't too much different in the behavior of the core and custom registry for now.

## Registry roles

> A user within an organization can have different roles, and therefore permissions, for each registry in their organization. Organization administrators can set a given user's permission.

Reviewer: ```suggestion
A user can have different roles, and therefore permissions, for each registry in their organization. Registry admins can set a given user's permission.
```

| Edit allowed types list for a registry                         | Update           |        |   X    |   X   |   X   |
| Edit custom registry name                                      | Update           |        |   X    |   X   |   X   |
| Delete a collection                                            | Delete           |        |   X    |   X   |   X   |
| Delete an automation                                           | Delete           |        |   X    |   X   |   X   |
| Unlink an artifact from a registry                             | Delete           |        |   X    |   X   |   X   |
| Edit accepted artifact types for a registry                    | Admin            |        |        |   X   |   X   |
| Change registry visibility (Organization or Restricted)        | Admin            |        |        |   X   |   X   |
| Add users to a registry                                        | Admin            |        |        |   X   |   X   |
| Assign or change a user's role in a registry                   | Admin            |        |        |   X   |   X   |

> ## Configure user roles in a registry

Reviewer: this is in general, but the settings page is accessible by everyone, but can only be changed by an admin. Sections that are only relevant for someone with registry admin access should be clearly marked (and ideally they should be consolidated in 1 area/guide so that a regular user doesn't have to keep scrolling through them when not relevant to them)

Author: I imply that only admins can edit/that this info pertains only to them, but I can make it more obvious by putting it in an info block?: 
<img width="558" alt="Screen Shot 2024-06-17 at 9 35 37 PM" src="https://github.com/wandb/docodile/assets/15202580/90d2e967-1849-498c-a079-59dd0845d7a9">

Reviewer: ooh I didn't follow what this screenshot refers to - is it to point out an info block and what that looks like? 

But anyways, yes, I think we should call out explicitly at the top of "Access Controls" section, and "Configure Registry Settings" section (we discussed [here](https://github.com/wandb/docodile/pull/689/files#r1648328099) that we will have 2 sections -> I think in an info block this is a good idea right after the header "Note - you must be a Registry admin to..."
